var/spack/repos/builtin/packages/openmp-extras/package.py

```diff
@@ -0,0 +1,319 @@
+            '-DDEVICE_LIBS_DIR={0}/amdgcn/bitcode'.format(devlibs_prefix),
+            '-DAOMP_STANDALONE_BUILD=0',
+            '-DDEVICELIBS_ROOT={0}/openmp-extras/rocm-device-libs'.format(src),
+            '-DOPENMP_TEST_C_COMPILER=$AOMP/bin/clang',
+            '-DOPENMP_TEST_CXX_COMPILER=$AOMP/bin/clang++',
+            '-DLIBOMPTARGET_AMDGCN_GFXLIST={0}'.format(gfx_list),
+            '-DLIBOMP_COPY_EXPORTS=OFF',
+            '-DHSA_INCLUDE={0}'.format(hsa_prefix),
+            '-DHSA_LIB={0}/lib'.format(hsa_prefix),
+            '-DHSAKMT_LIB={0}/lib'.format(hsakmt_prefix),
+            '-DHSAKMT_LIB64={0}/lib64'.format(hsakmt_prefix),
+            '-DCOMGR_INCLUDE={0}/include'.format(comgr_prefix),
+            '-DCOMGR_LIB={0}/lib'.format(comgr_prefix),
+            '-DLIBOMPTARGET_NVPTX_DEBUG=ON',
+            '-DOPENMP_ENABLE_LIBOMPTARGET=1',
+            '-DOPENMP_ENABLE_LIBOMPTARGET_HSA=1',
+            '-DLLVM_MAIN_INCLUDE_DIR={0}{1}'.format(src, llvm_inc),
+            '-DLLVM_INSTALL_PREFIX={0}'.format(openmp_extras_prefix),
+            '-DCMAKE_CXX_FLAGS=-g',
+            '-DCMAKE_C_FLAGS=-g'
+        ]
+
+        components['pgmath'] = [
+            '../openmp-extras/flang/runtime/libpgmath',
+            '-DLLVM_ENABLE_ASSERTIONS=ON',
+            '-DLLVM_CONFIG={0}/llvm-config'.format(bin_dir),
+            '-DCMAKE_CXX_COMPILER={0}/clang++'.format(bin_dir),
+            '-DCMAKE_C_COMPILER={0}/clang'.format(bin_dir),
+            '-DCMAKE_Fortran_COMPILER={0}/flang'.format(bin_dir),
+            '-DLLVM_TARGETS_TO_BUILD=AMDGPU;x86'
+        ]
+
+        components['flang'] = [
+            '../openmp-extras/flang',
+            '-DLLVM_ENABLE_ASSERTIONS=ON',
+            '-DLLVM_CONFIG={0}/llvm-config'.format(bin_dir),
+            '-DCMAKE_CXX_COMPILER={0}/clang++'.format(bin_dir),
+            '-DCMAKE_C_COMPILER={0}/clang'.format(bin_dir),
+            '-DCMAKE_Fortran_COMPILER={0}/flang'.format(bin_dir),
+            '-DLLVM_TARGETS_TO_BUILD=AMDGPU;x86',
+            '-DFLANG_OPENMP_GPU_AMD=ON',
+            '-DFLANG_OPENMP_GPU_NVIDIA=ON'
+        ]
+
+        components['flang-runtime'] = [
+            '../openmp-extras/flang',
+            '-DLLVM_ENABLE_ASSERTIONS=ON',
+            '-DLLVM_CONFIG={0}/llvm-config'.format(bin_dir),
+            '-DCMAKE_CXX_COMPILER={0}/clang++'.format(bin_dir),
+            '-DCMAKE_C_COMPILER={0}/clang'.format(bin_dir),
+            '-DCMAKE_Fortran_COMPILER={0}/flang'.format(bin_dir),
+            '-DLLVM_TARGETS_TO_BUILD=AMDGPU;x86',
+            '-DLLVM_INSTALL_RUNTIME=ON',
+            '-DFLANG_BUILD_RUNTIME=ON',
+            '-DOPENMP_BUILD_DIR={0}/spack-build-openmp/runtime/src'.format(src)
+        ]
+
+        build_order = [
+            "llvm", "aomp-extras", "openmp", "openmp-debug", "pgmath",
+            "flang", "flang-runtime"
+        ]
+
+        # Override standard CMAKE_BUILD_TYPE
+        std_cmake_args.remove("-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo")
```

Comment:
All cmake packages have a `build_type=RelWithDebInfo` variant. If someone changes this variant on the command line, it will no longer be RelWithDebInfo, so this isn't very robust. I guess you'll need to loop through and remove any args containing the string `CMAKE_BUILD_TYPE`.

Reply:
done